system_setup/server/controllers/configure.py

```diff
@@ -69,13 +66,88 @@ async def configure(self, *, context):
+                                 "-p", wsl_identity.password,
+                                 wsl_identity.username])
+                    run_command(["/usr/sbin/usermod", "-a",
+                                 "-c", wsl_identity.realname,
+                                 "-G", self.get_userandgroups(),
+                                 wsl_identity.username])
+            else:
+                wslconf_ad = self.model.wslconfadvanced.wslconfadvanced
+                config.update("WSL.automount.enabled",
+                              wslconf_ad.automount)
+                config.update("WSL.automount.mountfstab",
+                              wslconf_ad.mountfstab)
+                config.update("WSL.automount.root",
+                              wslconf_ad.custom_path)
+                config.update("WSL.automount.options",
+                              wslconf_ad.custom_mount_opt)
+                config.update("WSL.network.generatehosts",
+                              wslconf_ad.gen_host)
+                config.update("WSL.network.generateresolvconf",
+                              wslconf_ad.gen_resolvconf)
+                config.update("WSL.interop.enabled",
+                              wslconf_ad.interop_enabled)
+                config.update(
+                    "WSL.interop.appendwindowspath",
+                    wslconf_ad.interop_appendwindowspath)
+                config.update("ubuntu.GUI.followwintheme",
+                              wslconf_ad.gui_followwintheme)
+                config.update("ubuntu.GUI.theme",
+                              wslconf_ad.gui_theme)
+                config.update("ubuntu.Interop.guiintergration",
+                              wslconf_ad.legacy_gui)
+                config.update("ubuntu.Interop.audiointegration",
+                              wslconf_ad.legacy_audio)
+                config.update("ubuntu.Interop.advancedipdetection",
+                              wslconf_ad.adv_ip_detect)
+                config.update("ubuntu.Motd.wslnewsenabled",
+                              wslconf_ad.wsl_motd_news)
+
+            wslconf_base = self.model.wslconfbase.wslconfbase
+            config.update("WSL.automount.root",
+                          wslconf_base.custom_path)
+            config.update("WSL.automount.options",
+                          wslconf_base.custom_mount_opt)
+            config.update("WSL.network.generatehosts",
+                          wslconf_base.gen_host)
+            config.update("WSL.network.generateresolvconf",
+                          wslconf_base.gen_resolvconf)
+
             self.app.update_state(ApplicationState.DONE)
         except Exception:
             kw = {}
             self.app.make_apport_report(
                 ErrorReportKind.INSTALL_FAIL, "configuration failed", **kw)
             raise
 
+    def get_userandgroups(self):
+        usergroups_path = '/usr/share/subiquity/users-and-groups'
```

Comment:
hardcoding this path is not correct:
- it doesn’t work when running from git, as it doesn’t take the current branch file
- can’t we extract and factor out code from `users-and-groups`?

Reply:
> it doesn’t work when running from git, as it doesn’t take the current branch file

Not sure I get this question correctly, but it do have a function `resource_path('users-and-groups')`; but the problem is that is under WSL dyring the real run, `SUBIQUITY_ROOT` won't be set in the real run. technically saying right now the following lines below already did what you want to do:

https://github.com/canonical/subiquity/blob/2ac6f6d8b7de43343fc10d84636cb64e31fec2a4/system_setup/server/controllers/configure.py#L136-L139

> can’t we extract and factor out code from users-and-groups?

what do you mean by this?

Reply:
No problem, we can set (actually, we already do in the packaging we prepared with jibel `SUBIQUITY_ROOT`. I suggest that if it’s not set, you set it on server startup to a directory relative to the current file, that way, this will work when developping locally!